pkg/api/pod/util.go

```diff
@@ -151,7 +175,7 @@ func visitContainerSecretNames(container *api.Container, visitor Visitor) bool {
 // Transitive references (e.g. pod -> pvc -> pv -> secret) are not visited.
 // Returns true if visiting completed, false if visiting was short-circuited.
 func VisitPodConfigmapNames(pod *api.Pod, visitor Visitor) bool {
-	VisitContainers(&pod.Spec, func(c *api.Container) bool {
+	VisitContainers(&pod.Spec, DefaultContainers, func(c *api.Container, containerType ContainerType) bool {
```

Comment:
This is deciding for callers of VisitPodConfigmapNames that they should not visit containers disabled by feature gate.

A quick sweep of the uses of this method shows that containers that are feature-disabled but persisted should generally be included:
* kubelet configmap manager: this should set up management for references from all persisted containers
* noderestriction admission: this should prevent referencing configmaps in all containers
* node authorizer: this should honor persisted configmaps references